modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/direct/TransportGetDatabaseConfigurationAction.java

```diff
@@ -0,0 +1,124 @@
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.ingest.geoip.IngestGeoIpMetadata;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TransportGetDatabaseConfigurationAction extends TransportMasterNodeAction<
+    GetDatabaseConfigurationAction.Request,
+    GetDatabaseConfigurationAction.Response> {
+
+    @Inject
+    public TransportGetDatabaseConfigurationAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            GetDatabaseConfigurationAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            GetDatabaseConfigurationAction.Request::new,
+            indexNameExpressionResolver,
+            GetDatabaseConfigurationAction.Response::new,
+            EsExecutors.DIRECT_EXECUTOR_SERVICE
+        );
+    }
+
+    @Override
+    protected void masterOperation(
+        final Task task,
+        final GetDatabaseConfigurationAction.Request request,
+        final ClusterState state,
+        final ActionListener<GetDatabaseConfigurationAction.Response> listener
+    ) {
+        IngestGeoIpMetadata geoIpMeta = state.metadata().custom(IngestGeoIpMetadata.TYPE, IngestGeoIpMetadata.EMPTY);
+        if (geoIpMeta.getDatabases().isEmpty()) {
+            if (request.getDatabaseIds().length == 0) {
+                // you asked for all, and there are none, we return the none that there are
+                listener.onResponse(new GetDatabaseConfigurationAction.Response(List.of()));
+            } else {
+                // you asked for *something*, and there are none, we 404
+                listener.onFailure(
+                    new ResourceNotFoundException(
+                        "database configuration or configurations {} not found, no database configurations are configured",
+                        Arrays.toString(request.getDatabaseIds())
+                    )
+                );
```

Comment:
One thing I'm still not sure that I like about this is that it treats `GET /_ingest/geoip/database/` and `GET /_ingest/geoip/database/*` differently. Are we sure we like that? I'm not sure I care for the divergent behavior.

Reply:
I'm not married to it -- we have that behavior for SLM, so it's not without precedent, but it's not a detail that I'm super attached to. I'll simplify it out.

Reply:
How does 90a749ca4077e6ee68311b0f09ee7c1f3748743c strike you?

Reply:
I think that looks good to me.